### DISCUSSION TITLE
Set a thread name for spawned task

### DISCUSSION THREAD

**Discussion by manio** [tokio-rs/tokio#3356] [Q&A] [answered]
Hi. Standard threads can be named (https://doc.rust-lang.org/std/thread/#configuring-threads) which is very handy especially when monitoring (eg using htop) because every single thread can be named and monitored individually.
I miss something similar when doing task::spawn for a tokio async task.
Is it currently possible to set a name for the thread somehow? Am I missing something?

I know that I can set it for the whole pool when creating runtime with ```.thread_name("my-custom-name")``` but this can't be used for a individual task.

**Comment by Darksonn** [comment, marked as answer]
Tokio does not spawn a new thread for every task, so the name of a thread cannot be task-specific.

**Comment by manio** [comment]
@Darksonn I meant this for threaded_scheduler() only

**Reply by Darksonn** [reply]
The threaded scheduler doesn't spawn a new thread for every task either.

**Comment by wangjia184** [comment]
is it possible to set a name for the task so that the task name is visible in https://github.com/tokio-rs/console >?

**Reply by mox692** [reply]
I think you could use [`task::Builder::name`](https://docs.rs/tokio/latest/tokio/task/struct.Builder.html#method.name).

**Reply by JKearnsl** [reply]
as mox692 said:

```rust
#[macro_export] 
macro_rules! spawn_named {
    ($name:expr, $future:expr) => {
        tokio::spawn(async {
            tokio::task::Builder::new()
                .name($name)
                .spawn($future)
                .expect("failed to spawn task {}", $name)
                .await
            .expect(format!("task {} failed :(", $name))
        });
    };
}
```

🙂🙂🙂
